src/thermostate/plotting.py

```diff
@@ -58,10 +57,15 @@ def plot(self, x_axis, y_axis):  # pragma: no cover
         """Hold the place of a plot function that a child class must establish."""
         pass
 
-    def add_state(self, state, key=None):
+    def add_state(self, state, key=None, label = None):
         """Add a state to the self.states dictionary and plot it."""
         if key is None:
             key = repr(state)
+        if label:
```

Comment:
In principle, `label` could be an empty string or the integer zero, both of which evaluate to `False` in this context. Here, you want to check specifically for `None`:

```python
if label is not None:
```